datalad/metadata/metadata.py

```diff
@@ -234,7 +234,7 @@ def query_aggregated_metadata(reporton, ds, aps, recursive=False,
                 matching_subds = [(sub, sub) for sub in sorted(agginfos)
                                   # we already have the base dataset
                                   if (rpath == curdir and sub != curdir) or
-                                  sub.startswith(_with_sep(rpath))]
+                                  path_startswith(sub, rpath)]
```

Comment:
At least, formally here again. Although I'm not sure, whether equality is implicitly excluded anyway